widget/text.go

```diff
@@ -359,7 +359,13 @@ func lineBounds(text []rune, wrap fyne.TextWrap, maxWidth int, measurer func([]r
 		switch wrap {
 		case fyne.TextTruncate:
 			for {
-				if measurer(text[low:high]) <= maxWidth {
+				curWidth := measurer(text[low:high])
+				symbWidth := (curWidth / len(text[low:high]))
+				curWidth += symbWidth
```

Comment:
Rather than adding an arbitrary (text containing only 'i's, 'l's, or 'j's will give a smaller symbWidth than text containing only 'm's, or 'w's) symbol width, you could reduce maxWidth. Currently 'maxWidth := t.Size().Width' but it should probably be 'maxWidth := t.Size().Width - 2*theme.Padding()'

Reply:
Hi @stuartmscott, thanks for your reply!
>   Any data to back up this claim? The PR doesn't reduce the big O complexity so I can't see where your speed increase comes from

First of all, It is unfortunately visible to the eye.
Then I have made some tests, I just simply calculated loops count and got next results:

fyne.TextTruncate
Exists method
Text len: 1409 loops count: 1361 
Text len: 1441 loops count: 1393 
Text len: 1473 loops count: 1425
New method
Text len: 1409 loops count: 3 
Text len: 1441 loops count: 3 
Text len: 1473 loops count: 3

fyne.TextWrapBreak
Exists method
Text len: 1441 loops count: 17632 
Text len: 1505 loops count: 19253 
Text len: 1537 loops count: 20085
New method
Text len: 1441 loops count: 243 
Text len: 1473 loops count: 250 
Text len: 1473 loops count: 250

fyne.TextWrapWord
Exists method
Text len: 1377 loops count: 1676 
Text len: 1441 loops count: 1837 
Text len: 1473 loops count: 1918
New method
Text len: 1441 loops count: 82 
Text len: 1441 loops count: 82 
Text len: 1473 loops count: 82

> you could reduce maxWidth

Totally agree with you :) I have made the changes.